common/wuanalysis/anarule.cpp

```diff
@@ -135,14 +135,44 @@ class IoSkewRule : public AActivityRule
             stat_type timeAvgLocalExecute = activity.getStatRaw(StTimeLocalExecute, StAvgX);
 
             stat_type cost;
-            //If one node didn't spill then it is possible the skew caused all the lost time
             unsigned actkind = activity.getAttr(WaKind);
-            if ((actkind==TAKspillread||actkind==TAKspillwrite) && activity.getStatRaw(stat, StMinX) == 0)
+            if ((actkind==TAKspillread||actkind==TAKspillwrite) && (activity.getStatRaw(stat, StMinX) == 0))
+            {
+                //If one node didn't spill then it is possible the skew caused all the lost time
                 cost = timeMaxLocalExecute;
+                result.set(ANA_IOSKEW_RECORDS_ID, cost, "Uneven worker spilling is causing uneven %s time", category);
+            }
             else
+            {
+                bool fileSkew = false;
+                if (stat == StTimeDiskWriteIO)
```

Comment:
I think there are 3 cases worth separating.

i) No skew in size written   => Significant skew in IO performance
ii) skew in size written but no skew in record counts => Significant skew in record sizes
iii) skew in size written and record counts => Significant skew in number of record 

Skew in size